games/rtype/include/Components.hpp

```diff
@@ -0,0 +1,212 @@
+     * @struct Render
+     */
+    struct Render {
+    };
+
+    /**
+     * @brief Component for the collision
+     * @details The collision component is used to check if an entity is colliding with another entity
+     * @struct Collision
+     * @var origin: The origin of the collision rectangle
+     * @var end: The end of the collision rectangle
+     */
+    struct Collision {
+        std::vector<float> origin;
+        std::vector<float> end;
+    };
+
+    /**
+     * @brief Enum for the different status of a sound
+     * @details The sound status is used to know if a sound is playing, paused or stopped
+     * @enum SoundStatus
+     * @var PLAYING: The sound is playing
+     * @var PAUSED: The sound is paused
+     * @var STOPPED: The sound is stopped
+     */
+    enum SoundStatus { PLAYING, PAUSED, STOPPED };
+
+    /**
+     * @brief Component for the sound
+     * @details The sound component is different from the music component because music is looping
+     * @struct Sound
+     * @var path: The path to the sound
+     * @var volume: The volume of the sound
+     * @var status: The status of the sound
+     */
+    struct Sound {
+        std::string path;
+        float volume;
+        SoundStatus status;
+    };
+
+    /**
+     * @brief Component for the music
+     * @details The music component is different from the sound component because music is looping
+     * @struct Music
+     * @var path: The path to the music
+     * @var volume: The volume of the music
+     * @var isLooping: If the music is looping
+     * @var status: The status of the music
+     */
+    struct Music {
+        std::string path;
+        float volume;
+        bool isLooping;
+        SoundStatus status;
+    };
+
+    /**
+     * @brief Component for the texture
+     * @details The texture component is used to store the path to the texture and the texture itself
+     * @struct Texture
+     * @var path: The path to the texture
+     * @var sprite: A shared pointer to the texture
+     */
+    struct Texture {
+        std::string path;
+        std::shared_ptr<Otter::Graphic::Raylib::RaylibTexture> sprite;
+    };
+
+    /**
+     * @brief Component transform
+     * @details The transform component is used to store the position, rotation and scale of an entity
+     * @struct Transform
+     * @var position: A vector of float for the position of the entity
+     * @var rotation: The rotation of the entity
+     * @var scale: The scale of the entity
+     */
+    struct Transform {
+        std::vector<float> position;
+        float rotation;
+        float scale;
+    };
+
+    /**
+     * @brief Component for the velocity
+     * @details The velocity component is used to store the speed and acceleration of an entity
+     * @struct Velocity
+     * @var speed: The speed of the entity
+     * @var acceleration: The acceleration of the entity
+     */
+    struct Velocity {

```

Comment:
maybe that will work but becarful of the fact that we can put acceleration on a vector3 dimention